src/components/movie-preview-modal.jsx

```diff
@@ -28,9 +30,28 @@ const BlurredImage = styled.div`
   }
 `;
 
-export const MoviePreviewModal = ({ whiteButton, buttons, howToWatch }) => {
+export const MoviePreviewModal = ({
+  whiteButton,
+  userButtons,
+  publishButtons,
+  howToWatch,
+  selectedMovie,
+  closeModal,
+  refetchMovies,
+}) => {
   const [isWatchedClicked, setIsWatchedClicked] = useState(false);
   const [isWantClicked, setIsWantClicked] = useState(false);
+  const [updateMovieStatus, { data, loading, error }] =
+    useMutation(UPDATE_MOVIE_STATUS);
+
+  const approveMovie = (movieId) => {
+    updateMovieStatus({
+      variables: {
+        where: { id: movieId },
+        data: { status: "published" },
+      },
```

Comment:
Movie status updates and the modal opens to confirm if the movie is to be approved. Once approved, it should automatically update the list instead of needing a manual refresh.

Reply:
Done!